Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -1673,7 +1673,9 @@ static void emitTextureStore(FunctionDefinitionWriter* writer, AST::CallExpressi
         writer->stringBuilder().append(", "_s);
         writer->visit(*arrayIndex);
     }
-    writer->stringBuilder().append(')');
+    writer->stringBuilder().append(");\n"_s, writer->indent());
+    writer->visit(texture);
+    writer->stringBuilder().append(".fence()"_s);
```

Comment:
I suppose this is only needed for `read_write` storage textures, and not all textures which call write (i.e., write-only).

Is there a way to only insert the fence for `read_write` textures?